### PR TITLE
chore: Add personal GMMK ANSI keymap

### PR DESCRIPTION
**Pull request by andrewcharnley**
## Description

Personal GMMK ANSI keymap without gimmicks - designed for development. Specifics documented in readme.md

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gmmk/pro/ansi/keymaps/andrewcharnley/readme.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Description
+A keymap designed to be functional for development purposes - no flashing lights or gimmicks. PRTSC and DEL key in the top right corner.
+Solid backlight that changes red if CAPS is on and reverts back afterwards. Brightness retained in both cases.
+Backlight is disabled on suspend and restored upon wake-up.
+The 'Windows' key is a second FN key. This makes it more natural to use the encoder wheel (good luck trying to do it with one hand).
+
+Compile using `./bin/qmk compile -kb gmmk/pro/ansi -km andrewcharnley`
```
Comment (fauxpark) [inline keyboards/gmmk/pro/ansi/keymaps/andrewcharnley/readme.md:7]:
```suggestion
Compile using `qmk compile -kb gmmk/pro/ansi -km andrewcharnley`
```

File: keyboards/gmmk/pro/ansi/keymaps/andrewcharnley/keymap.c [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  // rgb_matrix_set_color_all(0,0,0);
+  rgb_matrix_set_suspend_state(true);
+}
+
+void suspend_wakeup_init_user (void) {
+
+  // rgb_matrix_indicators_advanced_user(0, 0);
+  rgb_matrix_set_suspend_state(false);
+}
```
Comment (fauxpark) [inline keyboards/gmmk/pro/ansi/keymaps/andrewcharnley/keymap.c:91]:
```suggestion
```
Shouldn't be necessary anymore.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/gmmk/pro/ansi/keymaps/andrewcharnley/keymap.c (hunk 2 above)
Comment (andrewcharnley) [reply keyboards/gmmk/pro/ansi/keymaps/andrewcharnley/keymap.c]: OK removed although I seemed to need it. I must have been unlucky when testing if you're sure of it.

Comment (fauxpark) [reply keyboards/gmmk/pro/ansi/keymaps/andrewcharnley/keymap.c]: Ah, maybe it is not necessary on develop, but that will be merging into master soon (and as a side note, you've opened this PR from your master branch).

**Pull request closed without merging**